src/dbnode/persist/fs/types.go

```diff
@@ -120,8 +120,10 @@ type DataFileSetReaderStatus struct {
 
 // DataReaderOpenOptions is options struct for the reader open method.
 type DataReaderOpenOptions struct {
-	Identifier  FileSetFileIdentifier
-	FileSetType persist.FileSetType
+	Identifier     FileSetFileIdentifier
+	FileSetType    persist.FileSetType
+	// OrderedByIndex enforces reading of series in the order of index (which is by series Id).
+	OrderedByIndex bool
```

Comment:
Prob better question for @robskillington, is this always true? In the tests above I see sorting logic for sorting by ID but is data written always already sorted by ID? Is this something we need to validate as we read in the read path?

Reply:
The index data here is stored lexiographically since it's built with Vellum that enforces the sorting, so we should be good here